core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
I wonder if we want to ban `new Random(int)` to make it harder for folks to do `new Random(System.currentTimeMillis)`. If so then `Randomness` is probably the right way to go. Otherwise I like `GOOD_FAST_HASH_SEED`.